File: android/app/src/main/java/dev/mobilewebcam/sender/model/StreamState.kt
package dev.mobilewebcam.sender.model

sealed interface StreamState {
    data object Idle : StreamState
    data object Connecting : StreamState
    data class Streaming(
        val session: StreamSession,
        val startedAtMillis: Long,
    ) : StreamState
    data object Stopping : StreamState
    data class Failed(val failure: StreamFailure) : StreamState
}

val StreamState.isSessionActive: Boolean
    get() = this is StreamState.Connecting ||
        this is StreamState.Streaming ||
        this is StreamState.Stopping

val StreamState.requiresStopConfirmation: Boolean
    get() = this is StreamState.Connecting ||
        this is StreamState.Streaming

File: sender/ios/Packages/CamBridgeCore/Sources/CamBridgeCore/Session/StreamState.swift
import Foundation

public enum StreamState: Equatable, Sendable {
    case idle
    case connecting(identity: SessionIdentity, configuration: StreamConfiguration)
    case streaming(identity: SessionIdentity, configuration: StreamConfiguration, mediaPort: Int)
    case stopping(identity: SessionIdentity)
    case failed(StreamFailure)

    public var diagnosticLabel: String {
        switch self {
        case .idle:
            "idle"
        case .connecting:
            "connecting"
        case .streaming:
            "streaming"
        case .stopping:
            "stopping"
        case .failed:
            "failed"
        }
    }
}
